File: Daily/PY/Leetcode1603-设计停车系统.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.big,self.mediu,self.small = big,medium,small

    def addCar(self, carType: int) -> bool:
        if carType == 1:
            if(self.big > 0):
                self.big -= 1
                return True
        elif carType == 2:
            if self.mediu > 0:
                self.mediu -= 1
                return True
        else:
            if self.small > 0:
                self.small -= 1
                return True
        return False

File: Leetcode/sbgo/Leetcode1603-设计停车系统.go
package sbgo

type ParkingSystem struct {
	big, medium, small int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{big: big, medium: medium, small: small}
}

func (this *ParkingSystem) AddCar(carType int) bool {
	if carType == 1 {
		if this.big > 0 {
			this.big--
			return true
		}
	} else if carType == 2 {
		if this.medium > 0 {
			this.medium--
			return true
		}
	} else {
		if this.small > 0 {
			this.small--
			return true
		}
	}
	return false
}
